quinn-proto/src/shared.rs

```diff
@@ -281,40 +280,13 @@ impl Default for TransportConfig {
     }
 }
 
-impl TransportConfig {
-    pub(crate) fn validate(&self) -> Result<(), ConfigError> {
-        if let Some((name, _)) = [
-            ("stream_window_bidi", self.stream_window_bidi),
-            ("stream_window_uni", self.stream_window_uni),
-            ("receive_window", self.receive_window),
-            ("stream_receive_window", self.stream_receive_window),
-            ("idle_timeout", self.idle_timeout),
-        ]
-        .iter()
-        .find(|&&(_, x)| x > VarInt::MAX.into_inner())
-        {
-            return Err(ConfigError::VarIntBounds(name));
-        }
-        if self.crypto_buffer_size < 4096 {
-            return Err(ConfigError::IllegalValue(
-                "crypto_buffer_size must be at least 4096",
-            ));
-        }
-        if self.idle_timeout != 0 && u64::from(self.keep_alive_interval) >= self.idle_timeout {
-            warn!(
-                "keep-alive interval {} is ineffective due to lower idle timeout {}",
-                self.keep_alive_interval, self.idle_timeout
-            );
-        }
-        Ok(())
-    }
-}
-
 /// Global configuration for the endpoint, affecting all connections
 ///
 /// Default values should be suitable for most internet applications.
-#[derive(Clone)]
-pub struct EndpointConfig {
+pub struct EndpointConfig<S>
```

Comment:
So this is where we lose `Clone` on `EndpointConfig`, which I think I added for hyper compat... I think that's the reason I moved `HmacKey` et al out of here in the first place.

Reply:
I solved this by wrapping the `HmacKey`s in `Arc` and adding an explicit `Clone` impl since `derive(Clone)` is pretty dumb about type parameters.